app/celery/tasks.py

```diff
@@ -245,21 +249,27 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Any],
         else:
             reply_to_text = template.get_reply_to_text()  # type: ignore
 
-        notification["reply_to_text"] = reply_to_text
-        notification["service"] = service
-        notification["key_type"] = notification.get("key_type", KEY_TYPE_NORMAL)
-        notification["template_id"] = template.id
-        notification["template_version"] = template.version
-        notification["recipient"] = notification.get("to")
-        notification["personalisation"] = notification.get("personalisation")
-        notification["notification_type"] = SMS_TYPE
-        notification["simulated"] = notification.get("simulated", None)
-        notification["api_key_id"] = notification.get("api_key", None)
-        notification["created_at"] = datetime.utcnow()
-        notification["job_id"] = notification.get("job", None)
-        notification["job_row_number"] = notification.get("row_number", None)
+        notification: VerifiedNotification = {
+            **_notification,  # type: ignore
+            "notification_id": notification_id,
+            "reply_to_text": reply_to_text,
+            "service": service,
+            "key_type": _notification.get("key_type", KEY_TYPE_NORMAL),
+            "template_id": template.id,
+            "template_version": template.version,
+            "recipient": _notification.get("to"),
+            "personalisation": _notification.get("personalisation"),
+            "notification_type": SMS_TYPE,
+            "simulated": _notification.get("simulated", None),
+            "api_key_id": _notification.get("api_key", None),
+            "created_at": datetime.utcnow(),
+            "job_id": _notification.get("job", None),
+            "job_row_number": _notification.get("row_number", None),
+        }
+
         verified_notifications.append(notification)
-        notification_id_queue[notification_id] = notification.get("queue")
+        # bug: notification does not have "queue" key
+        notification_id_queue[notification_id] = notification.get("queue")  # type: ignore
```

Comment:
I thought I put a comment here 🤦🏽 this is coming from the signed notification - the notification table has a "queue_name" -> we are sending it through the process_rows function https://github.com/cds-snc/notification-api/blob/main/app/celery/tasks.py#L169

Reply:
Okay thanks for pointing that out! I hadn't noticed that function. I'll add `queue` to the `NotificationDictToSign` type